### ISSUE TITLE
Properly inform user about expired session

### ISSUE THREAD

**Issue by Leitsi** [HSLdevcom/jore4#1291]
Happens in development environment when restarting the UI dependencies, but possibly in production as well.
When trying to use Jore4 UI with an expired session, currently the UI just loads partially and is broken. To fix this the user has to go to front page where login is prompted. Instead, we should probably have a some kind of prompt in a modal or redirect.

Discuss with design if needed.

- [x] #1569

**Comment by juhakolari** [comment]
MVP could be just automatic refresh, unless it leads to data loss. However, one could take note of the following WCAG requirements as well:

Success criterion (Level A) for [2.2.1. Timing adjustable](https://www.w3.org/WAI/WCAG21/Understanding/timing-adjustable.html) requires that the user should be able warned of timeout at least 20 seconds before it occurs _or_  the time limit is longer than 20 hours. 

Success criterion (Level AAA) for [2.2.6. Timeouts](https://www.w3.org/WAI/WCAG21/Understanding/timeouts.html) reads: _"Users are warned of the duration of any [user inactivity](https://www.w3.org/WAI/WCAG21/Understanding/timeouts.html#dfn-user-inactivity) that could cause data loss, unless the data is preserved for more than 20 hours when the user does not take any actions."_

**Comment by niknii-vincit** [comment]
Additional behaviour note: you can save a link to a line/route, and go to it with the expired session and it will be an empty view.

**Comment by niknii-vincit** [comment]
Managed to crash the application if you try to upload a timetable without logging in, was about to create a ticket, but I think this ticket will prevent/handle this issue as well? 
![image](https://github.com/HSLdevcom/jore4/assets/135983800/c62f0b9c-4998-4cb8-bb95-0c01f89b94c9)

**Comment by HenrikHartiala** [comment]
Yes, all pages (exlcuding landing page) is going to require authentication

**Comment by HenrikHartiala** [comment]
Reopening this ticket since I've linked a wrong ticket to my pull request. This is still to be implemented

**Comment by niknii-vincit** [comment]
![Image](https://github.com/HSLdevcom/jore4/assets/135983800/2946d83e-d692-4daf-851f-815bb7981173)

Got this a few times today, when trying to import a timetable on a shoddy network. What seemed to be the authentication, was pretty inconsistent and surprisingly network heavy and failed a few times. I probably didn't do a rebase before doing a final test and pushing to remote... So this ticket might have solved the issue I had, and we probably don't need to worry about shoddy network?
